### PR TITLE
Plasma shape in polar coordinates

### PR DESCRIPTION
**Pull request by awlishar**
In the previous versions, the plasma shape was given in cartesian coordinates that only included the top half of the boundary. Now, the boundary shape is given in polar coordinates rho(theta), which supports full boundaries.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,13 +5,13 @@ KARHU is a machine learning surrogate model for the ideal peeling-ballooning MHD
...
+See publication refering to KARHU v1.0.0: [https://doi.org/10.1063/5.0282085](https://doi.org/10.1063/5.0282085)
 
 
 The training data is available in folder ´data´. Future datasets will be hosted on Zenodo.
 
 
-In `v1`, `KARHU` takes as **ordered** inputs, i.e., `forward(P, Q, RBPHI, ZBNDRY, BMAG, RMAG)`, where `P, Q, RBPHI` are the pressure, q (safety factor) and poloidal current functions of $\psi$ defined on a uniform $\psi_N^2$ grid of `64` points. `ZBNDRY` is the Z-coordinates of the LCFS defined on a unifrom R-grid from `R_min, R_max` with also `64` points. Finally, `BMAG, RMAG` are the values of the toroidal field and major radius at the magnetic axis. 
+In `v1`, `KARHU` takes as **ordered** inputs, i.e., `forward(P, Q, RBPHI, RHOBNDRY, BMAG, RMAG)`, where `P, Q, RBPHI` are the pressure, q (safety factor) and poloidal current functions of $\psi$ defined on a uniform $\psi_N^2$ grid of `64` points. `RHOBNDRY` is the rho-coordinates (polar coordinate system) of the LCFS defined on a unifrom theta-grid from `0, 2 pi` with `128` points. Finally, `BMAG, RMAG` are the values of the toroidal field and major radius at the magnetic axis.
```
Comment (fusionby2030) [inline README.md]: this should be now `v2` instead of `v1`.

File: model/diii-d/scaling_params.json [hunk 2]
```diff
@@ -12,8 +12,8 @@
         1.0832818746566772
     ],
     "shape": [
-        0.026455655694007874,
-        1.8997050523757935
+        0.9968876838684082,
```
Comment (fusionby2030) [inline model/diii-d/scaling_params.json:15]: shape is a bit of an arbitrary name, but if we have proper documentation (like we do with the tests and inference.ipynb) then this is explainable.

File: model/jet_2H/pred_vs_true_testset.png
Comment (fusionby2030) [inline model/jet_2H/pred_vs_true_testset.png:1]: Do we have a script which generates this figure? That would be even better :D.

File: model/jet_2H/scaling_params.json [hunk 3]
```diff
@@ -30,5 +34,9 @@
     "b_mag": [
         1.03132164478302,
         5.352698802947998
+    ],
+    "beta_n": [
```
Comment (fusionby2030) [inline model/jet_2H/scaling_params.json:38]: grrrrrrr

File: src/karhu/cli/inference_from_helena.py [hunk 4]
```diff
@@ -2,28 +2,19 @@
...
 import argparse 
-from karhu.utils_helena import get_model_input
+from karhu.utils_helena import load_from_helena
 from karhu.models import load_model
 from karhu.utils_input import scale_model_input, scale_model_output
 
 import torch
 
-WP     = torch.float32 #TODO: will this change in future? 
+WP     = torch.float32  # TODO: will this change in future?
```
Comment (fusionby2030) [inline src/karhu/cli/inference_from_helena.py:11]: The # TODO here is actually more for the user. We should note to the users which format the model is trained in, but you can in principle cast it to whatever precision. Maybe we have to document the performance hit?

File: src/karhu/common.py [hunk 5]
```diff
@@ -41,3 +41,151 @@ def forward_pass(x: list[torch.tensor], model, scaling_params) -> torch.tensor:
...
+        and the function mirrors it to produce a full 0-2pi contour.
+    - If asymmetric: uses the full input directly.
+
+    Parameters
+    ----------
+    r_vals : array_like
+        R (major radius) coordinates of the boundary.
+    z_vals : array_like
+        Z (vertical) coordinates of the boundary.
+
```
Comment (fusionby2030) [inline src/karhu/common.py:62]: add doc string about `r0`, `z0` and `amin`. Once you transform the boundary into into the `MISHKA/KARHU` normalisation space, then the $r_0$, $z_0$ and $a_min$ are not needed right (i.e., there defaults)? But if you are doing the conversion in real space then one has to move into (r0, z0) centered coordinate space to do this transformation.

File: src/karhu/models.py [hunk 6]
```diff
@@ -20,14 +20,23 @@ def load_model(model_dir: str) -> tuple[torch.nn.Module, dict[str, np.ndarray]]:
         scaling_params = json.load(f)
 
     # Load model
-    model = CNN_gmax()
+    model = CNN_gmax(
+        conv_input_sizes=[64, 64, 64, 128],
```
Comment (fusionby2030) [inline src/karhu/models.py:24]: Hm, I think these should come from the `config` file, as leaving it "flexible" for the user doesn't really make sense to me.

Comment (fusionby2030) [reply src/karhu/models.py:24]: Either that or we should just use torchscript?

**Review by fusionby2030 — COMMENTED** [review]
Nice! You could also delete in the `common` the `forward_pass` function.

File: model/diii-d/scaling_params.json (hunk 2 above)
Comment (awlishar) [reply model/diii-d/scaling_params.json:15]: Not for this PR to change, but I could change it to "boundary" in the PR where I anyway change the scaling_params.json file.

File: model/jet_2H/pred_vs_true_testset.png
Comment (awlishar) [reply model/jet_2H/pred_vs_true_testset.png:1]: In the karhu-training repo :)

File: model/jet_2H/scaling_params.json (hunk 3 above)
Comment (awlishar) [reply model/jet_2H/scaling_params.json:38]: XD

File: src/karhu/models.py (hunk 6 above)
Comment (awlishar) [reply src/karhu/models.py:24]: Agreed, will include in another PR!

File: src/karhu/common.py (hunk 5 above)
Comment (awlishar) [reply src/karhu/common.py:62]: Done!
